src/daft-local-execution/src/sinks/aggregate.rs

```diff
@@ -38,44 +30,22 @@ impl AggregateSink {
 impl BlockingSink for AggregateSink {
     #[instrument(skip_all, name = "AggregateSink::sink")]
     fn sink(&mut self, input: &Arc<MicroPartition>) -> DaftResult<BlockingSinkStatus> {
-        if let AggregateState::Accumulating(parts) = &mut self.state {
-            parts.push(input.clone());
-            Ok(BlockingSinkStatus::NeedMoreInput)
-        } else {
-            panic!("sink must be in Accumulating phase")
-        }
+        self.state.push(input.clone());
+        Ok(BlockingSinkStatus::NeedMoreInput)
     }
 
     #[instrument(skip_all, name = "AggregateSink::finalize")]
-    fn finalize(&mut self) -> DaftResult<()> {
-        if let AggregateState::Accumulating(parts) = &mut self.state {
-            assert!(
-                !parts.is_empty(),
-                "We can not finalize AggregateSink with no data"
-            );
-            let concated =
-                MicroPartition::concat(&parts.iter().map(|x| x.as_ref()).collect::<Vec<_>>())?;
-            let agged = concated.agg(&self.agg_exprs, &self.group_by)?;
-            self.state = AggregateState::Done(Arc::new(agged));
-            Ok(())
-        } else {
-            panic!("finalize must be in Accumulating phase")
```

Comment:
This ensure that finalize is never called twice and the state is what we expect.